src/log_prob.jl

```diff
@@ -0,0 +1,438 @@
+
+        # multiply by image's gain for this pixel
+        rate     = pixel_rate * tile.iota_vec[pixel.h]
+        pixel_ll = logpdf(Poisson(rate[1]), round(Int, this_pixel))
+        ll += pixel_ll
+    end
+    return ll
+end
+
+
+###########################################################################
+# priors - for computing prior log probs of star/gal parameters           #
+###########################################################################
+
+
+type StarPrior
+    brightness::LogNormal
+    color_component::Categorical
+    colors::Vector{MvNormal}
+end
+
+
+type GalaxyPrior
+    brightness::LogNormal
+    color_component::Categorical
+    colors::Vector{MvNormal}
+    gal_scale::LogNormal
+    gal_ab::Beta
+    gal_fracdev::Beta
+end
+
+
+type Prior
+    is_star::Bernoulli
+    star::StarPrior
+    galaxy::GalaxyPrior
+end
+
+
+"""
+Construct a `Prior` object, contains StarPrior and GalaxyPrior objects
+"""
+function construct_prior()
+    pp = load_prior()
+    star_prior = StarPrior(
+        LogNormal(pp.r_mean[1], pp.r_var[1]),
+        Categorical(pp.k[:,1]),
+        [MvNormal(pp.c_mean[:,k,1], pp.c_cov[:,:,k,1]) for k in 1:2])
+
+    gal_prior = GalaxyPrior(
+        LogNormal(pp.r_mean[2], pp.r_var[2]),
+        Categorical(pp.k[:,2]),
+        [MvNormal(pp.c_mean[:,k,2], pp.c_cov[:,:,k,2]) for k in 1:2],
+        LogNormal(0, 10),
+        Beta(1, 1),
+        Beta(1, 1))
+
+    return Prior(Bernoulli(.5), star_prior, gal_prior)
+end
+
+
+"""
+Return the log prior probability of a (brightness,color) pair based on
+prior probability objects constructed by `construct_prior()`
+"""
+function color_logprior(brightness::Float64,
+                        colors::Vector{Float64},
+                        prior::Prior,
+                        is_star::Bool)
+    subprior      = is_star ? prior.star : prior.galaxy
+    ll_brightness = logpdf(subprior.brightness, exp(brightness))
+    ll_component  = [logpdf(subprior.colors[k], colors) for k in 1:2]
+    ll_color      = logsumexp(ll_component + log(subprior.color_component.p))
+    return ll_brightness + ll_color
+end
+
+
+function shape_logprior(gal_shape::Vector{Float64}, prior::Prior)
+    # position and gal_angle have uniform priors--we ignore them
+    gdev, gaxis, gangle, gscale = gal_shape
+    ll_shape = 0.
+    ll_shape += logpdf(prior.galaxy.gal_scale, gscale)
+    ll_shape += logpdf(prior.galaxy.gal_ab, gaxis)
+    ll_shape += logpdf(prior.galaxy.gal_fracdev, gdev)
+    return ll_shape
+end
+
+
+#################################################################
+# transformation of fluxes = (u, g, r, i, z) in nanomaggies to  #
+# (bright,color) pairs                                          #
+#################################################################
+
+
+"""
+Translate between fluxes (nmgy rates) and colors (on which
+LogNormal priors are placed)
+"""
+function fluxes_to_colors(fluxes::Vector{Float64})
```

Comment:
Let's add a unit test that verifies that this and ```colors_to_fluxes``` match ```SourceBrightness``` when the variational variance is very small.

Extra nice might be to put these functions in ```source_brightness.jl``` so it's clear that if we change one we should change the other.